graylog2-server/src/main/java/org/graylog2/inputs/codecs/GelfCodec.java

```diff
@@ -218,6 +222,47 @@ public Message decode(@Nonnull final RawMessage rawMessage) {
         return message;
     }
 
+    private void validateGELFMessage(JsonNode jsonNode, UUID id, ResolvableInetSocketAddress remoteAddress) {
+        final String prefix = "GELF message <" + id + "> " + (remoteAddress == null ? "" : "(received from <" + remoteAddress + ">) ");
+
+        final JsonNode hostNode = jsonNode.path("host");
+        if (hostNode.isMissingNode()) {
+            log.warn(prefix + "is missing mandatory \"host\" field.");
+            return;
```

Comment:
Shouldn't we check for the `short_message` / `message` fields first before returning? As it is now, a missing `host` field would skip the `short_message` / `message` field checks.